### PR TITLE
Support property in fake provider

### PR DESCRIPTION
**Pull request by shuheiktgw**
## Problem Statement

SSIA

## Related Issue

Relates to https://github.com/external-secrets/external-secrets/issues/1655

## Proposed Changes

Previous discussion exists at https://github.com/external-secrets/external-secrets/pull/1285. It's been closed so I'd like to try implementing the feature in this new PR!

## Checklist

- [x] I have read the [contribution guidelines](https://external-secrets.io/latest/contributing/process/#submitting-a-pull-request)
- [x] All commits are signed with `git commit --signoff`
- [x] My changes have reasonable test coverage
- [x] All tests pass with `make test`
- [x] I ensured my PR is ready for review with `make reviewable`

### CODE REVIEW

**Comment by shuheiktgw** [comment]
Hmm, the CI/check-diff failure does not seem to be caused by the change I made in this PR, does it? 🤔

File: docs/api/spec.md [hunk 1]
```diff
@@ -5105,7 +5105,7 @@ External Secrets meta/v1.SecretKeySelector
 <td><p>Error indicates that there is a misconfiguration.</p>
 </td>
 </tr><tr><td><p>0</p></td>
-<td><p>Ready indicates that the client is configured correctly
+<td><p>Ready indicates that the client is confgured correctly
```
Comment (moolen) [inline docs/api/spec.md:5108]: I think this is it;
```suggestion
<td><p>Ready indicates that the client is configured correctly
```

**Review by moolen — APPROVED** [review]
LGTM, thank you for this :heart:

**Comment by shuheiktgw** [comment]
Thank you for your review!

**Pull request merged**